adcc/ExcitedStates.py

```diff
@@ -582,7 +643,78 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
             ret += separator
             formatted = vector_format.format(vec).replace("\n", " |\n| ")
             ret += "| " + formatted + " |\n"
-            if i != len(self.excitation_vectors) - 1:
+            if i != len(self.excitation_vector) - 1:
                 ret += "\n"
                 ret += separator
         return ret[:-1]
+
+    @property
+    def excitations(self):
+        """
+        Provides a list of Excitations, i.e., a view to all individual
+        excited states and their properties. Still under heavy development.
+        """
+        excitations = []
+        for i in range(self.size):
+            e = Excitation(self, index=i, method=self.method)
+            for key in dir(self):
+                if key == "excitations":
+                    continue
+                if "__" in key or key.startswith("_"):
+                    continue  # skip "private" fields
+                if not hasattr(type(self), key):
+                    continue
+                if not isinstance(getattr(type(self), key), property):
+                    continue
+
+                fget = getattr(type(self), key).fget
+                if not hasattr(fget, "__excitation_property"):
+                    continue
+                # Extract the kwargs passed to mark_excitation_property
+                kwargs = getattr(fget, "__excitation_property").copy()
+
+                def get_parent_property(self, key=key, kwargs=kwargs):
+                    return getattr(self.parent_state, key)[self.index]
+
+                setattr(type(e), key, property(get_parent_property))
+
+                transform = kwargs.pop("transform_to_ao", False)
+                if transform:
+                    def get_parent_property_transform(self, key=key):
+                        matrix = getattr(self.parent_state, key)[self.index]
+                        density_ao = sum(matrix.to_ao_basis())
+                        return density_ao.to_ndarray()
+                    setattr(type(e), key + "_ao",
+                            property(get_parent_property_transform))
+            excitations.append(e)
```

Comment:
I would actually put this into the Excitation constructor as well. What do you think?

Reply:
Let me try quickly if it works.

Reply:
Well... I guess then we have a cyclic import? `ExcitedStates.py` imports `Excitation` and the other way around?

Reply:
Why? you never need the class definition, you only need the instance variable and `dir` and `getattr` and such magic. Maybe I miss something, but I don't see right now why `Excitation`  needs to know anything from `ExcitedStates.py`.

Reply:
You're right, I made a wrong assumption about the code 😉 
I like the split version much better now 👍